docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+# DRAFT NUMBER 2
+
+# Introduction
+
+The Babbage->Conway transition on an earlier iteration of SanchoNet revealed a bug in the Consensus layer by which the Hard Fork Combinator transitioned to Conway because the relevant Babbage protocol parameter update proposal received sufficient votes but then that same proposal wasn't actually enacted: the major protocol version remained unchanged even after Conway started.
+
+Troubleshooting this bug and moreover determining how to fix it in such a way that makes similar mistakes less likely in the future has lead to the following mini-framework for understanding the HFC's behavior on a particular chain.
+
+So many dimensions arose from investiating such a relatively simple misbehavior for two reasons.
+First, the mechanisms implementing these policies are---at least in the current implementation---somewhat interdependent.
+Second, the HFC (including its instantiation for Cardano) remains underdocumented since back when it was originaly developed under time-pressure, and our team no longer includes its inventor.
+The Consensus Team learns new things and unfortunately re-learns some old things every time they dive in to alter it.
+(Optimistically, Nicolas Frisby feels they currently understand it now better than they have ever before.)
+
+## Dimensions
+
+The following discrete variables characterize dimensions of the HFC design space.
+In this context, the _determination_ is the `singleEraTransition` method: it's how an era's ledger state determines when the era ends.
+
+- DST - Determination is Sensitive to Ticks  - whether ticking the ledger state can change its determination of the end of its era: Sensitive vs Insensitive
+
+- CCT - Criss-Crossed Ticks - whether an era's tick rules might be used to tick across slots that are in _other_ eras: Allowed vs Disallowed
+
+- PIL - Potential to Ignore the Ledger - whether the HFC can ignore the ledger's hard fork-related governance: Possible vs Impossible
+
+- DSR - Determination is Sensitive to Rollback - for chains C and D where C < D and len(C) <= len(C /\ D) + k, whether the current era of C can have a determined end on C but an undetermined or different end on D: Sensitive vs Insensitive
+
+- ZEE - Zero Epoch Eras - whether an era (in its entirety) can contain zero epochs: Allowed vs Disallowed
+
+- ZBE - Zero Block Eras - whether an era (in its entirety) can contain zero blocks: Allowed vs Disallowed
+
+- QSR - Queries are Sensitivity to Rollback - same as DSR but for HFC-related Node-To-Client queries instead of for validity of blocks: Sensitive vs Insensitive
+
+Note the following relations.
+
+- ZEE=Allowed implies ZBE=Allowed.
+- DSR=Insensitive implies QSR=Insensitive.
```

Comment:
I don't get this. What about queries that are not related to era transitions?

Reply:
> What about queries that are not related to era transitions?

I wasn't considering them at all here. So this document doesn't constraint them at all.

I should say that explicitly in the definition of QSR then.